src/fields/struct_field.rs

```diff
@@ -87,9 +90,18 @@ impl StructField {
                     Versionize::serialize(element, writer, version_map, app_version)?;
                 }
             },
-            syn::Type::Path(_) => quote! {
-                Versionize::serialize(&copy_of_self.#field_ident, writer, version_map, app_version)?;
-            },
+            syn::Type::Path(_) => {
+                if self.packed_struct {
+                    quote! {
+                        let copy = copy_of_self.#field_ident;
+                        Versionize::serialize(&copy, writer, version_map, app_version)?;
+                    }
+                } else {
+                    quote! {
+                        Versionize::serialize(&copy_of_self.#field_ident, writer, version_map, app_version)?;
+                    }
+                }
```

Comment:
How are the two branches different here? From what I can tell, compiled code would look the same in both cases...

Reply:
You can't reference struct members that are packed. By copying to the stack and then referencing that you avoid the problem (since you're not referencing the packed struct member.)

Reply:
See https://rust-lang.github.io/rfcs/1240-repr-packed-unsafe-ref.html but tl;dr alignment requirements can't be guaranteed if referencing packed structs.